### PR TITLE
First version of upgrading section

### PR DESCRIPTION
**Pull request by windsource**
Issue-Id: #184

Issues: #

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [ ] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: doc/docs/usage/upgrading/v0_2_to_v0_3.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Upgrading v0.2 to v0.3
+
+When upgrading from v0.2 to v0.3, the installation script simply needs to be run again. However, due to breaking changes, some manual adjustments will be required for existing configurations and workloads.
+
+`CompleteState`
+
+* `currentState ` has been renamed to `desiredState`
```
Comment (lingnoi) [inline doc/docs/usage/upgrading/v0_2_to_v0_3.md:7]:
```suggestion
* `currentState` has been renamed to `desiredState`
```

File: doc/docs/usage/upgrading/v0_2_to_v0_3.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+`State`
+
+* New mandatory field `apiVersion`
+* The field `restart` has been renamed to `restartPolicy` and the value is not boolean but an enum.
+
+## API
+
+Application using the control interface or communication directly with the Ankaios server need to be adapted.
+
+A new `ToServer` message, `Request`, has been introduced.
```
Comment (maturar) [inline doc/docs/usage/upgrading/v0_2_to_v0_3.md:20]: Is the `ToServer` new message? Or just a renaming of `StateChangeRequest`?

File: doc/docs/usage/upgrading/v0_2_to_v0_3.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+* The field `restart` has been renamed to `restartPolicy` and the value is not boolean but an enum.
+* `configs` and `cronjobs` have been removed.
+
+`Workload`
+
+* `accessRights` and `updateStrategy` have been removed.
+
+## API
+
+Application using the control interface or communication directly with the Ankaios server need to be adapted.
```
Comment (krucod3) [inline doc/docs/usage/upgrading/v0_2_to_v0_3.md:23]:
```suggestion
Application using the control interface or communicating directly with the Ankaios server (custom CLIs) need to be adapted.
```

File: doc/docs/usage/upgrading/v0_2_to_v0_3.md (hunk 2 above)
Comment (krucod3) [reply doc/docs/usage/upgrading/v0_2_to_v0_3.md:25]: I think it was meant as message type.

**Review by krucod3 — COMMENTED** [review]
Just a couple of suggestions.

**Review by krucod3 — APPROVED** [review]
LGTM 👍

File: doc/docs/usage/upgrading/v0_2_to_v0_3.md (hunk 2 above)
Comment (christoph-hamm) [reply doc/docs/usage/upgrading/v0_2_to_v0_3.md]: I have now also described the renames and clarified the sentences about the new types, Response and Request.

File: doc/docs/usage/upgrading/v0_2_to_v0_3.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+* The field `restart` has been renamed to `restartPolicy` and the value is not boolean but an enum.
+
+## API
+
+Application using the control interface or communication directly with the Ankaios server need to be adapted.
+
+A new `ToServer` message, `Request`, has been introduced.
+The two message `UpdateStateRequest` and `CompleteStateRequest` have been moved to the new `Request` message.
+
+A new `FromServer` message, `Response`, has been introduced.
```
[the remark this answers is not in the document]
Comment (christoph-hamm) [reply doc/docs/usage/upgrading/v0_2_to_v0_3.md]: I have clarified this sentence.

**Review by maturar — APPROVED** [review]
LGTM 👍

**Pull request merged**